### PR TITLE
Use boost's predefined cpp integer types

### PR DESCRIPTION
**Pull request by quangloc99**
Part of #15435/#15464.

This should help _fix_ a minor issue in `test/tools/yulInterpreter/EVMInstructionInterpreter.cpp`. Here the type `u512` is defined with `MaxBits` greater than `MinBits`. This does not cause any error since boost use the max of `MaxBits` and MinBits` for `void` allocator. Though it should be fixed to avoid confusion.

The reference for the types is here https://www.boost.org/doc/libs/1_84_0/libs/multiprecision/doc/html/boost_multiprecision/tut/ints/cpp_int.html

### CODE REVIEW

**Review by cameel — COMMENTED** [review]
Looks good, but needs a rebase before we can merge this (I'd actually rebase myself, but I see that the option to let maintainers push to a PR was not enabled here).

We generally do not merge the `develop` branch into PRs to update them because it makes history hard to follow. We always rebase branches on top of `develop` instead.

**Comment by quangloc99** [comment]
I have rebased this branch.
I see that the `Allow edits by maintainers` enabled on my side tho :pray: .

**Review by clonker — APPROVED** [review]

**Review by cameel — APPROVED** [review]

**Comment by cameel** [comment]
> I see that the `Allow edits by maintainers` enabled on my side tho 🙏 .

Oh, I assumed it wasn't because github did not show the button to update the PR. But I guess it makes sense. The branch was technically up to date.

**Pull request merged**
